src/pudl/analysis/record_linkage/embed_dataframe.py

```diff
@@ -108,6 +108,42 @@ def embed_dataframe_graph(df: pd.DataFrame) -> FeatureMatrix:
     return embed_dataframe_graph
 
 
+def dataframe_cleaner_factory(
+    name_prefix: str, vectorizers: dict[str, ColumnVectorizer]
+):
+    """Return a configured op graph to clean an input dataframe."""
+
+    @op(name=f"{name_prefix}_train")
+    def train_dataframe_cleaner(df: pd.DataFrame):
+        """Train :class:`sklearn.compose.ColumnTransformer` on input."""
+        column_transformer = ColumnTransformer(
+            transformers=[
+                (name, column_transform.as_pipeline(), column_transform.columns)
+                for name, column_transform in vectorizers.items()
+            ],
+        ).set_output(transform="pandas")
```

Comment:
I kind of think this slows down performance but maybe I'm wrong and my computer is just moving slow. I could instead have it output a numpy array (default) and then create a dataframe from it with assigned column names.